src/effekt/benchmark/queens.effekt

```diff
@@ -0,0 +1,102 @@
+import src/effekt/benchmark
+import src/effekt/cliRunner
+import mutable/array
+
+type Return[T] {
+  Continue()
+  BreakWith(value: T)
+}
+
+def loopWithReturn[T](from: Int, to: Int, finally: T){ fnc: Int => Return[T] } : T = {
+  if (from == to) {
+    finally
+  } else {
+    val fncVal = fnc(from);
+    val output: T = fncVal match {
+      case BreakWith(value) => value //early return
+      case Continue() => loopWithReturn(from+1,to, finally){ fnc };
+    }
+    output
+  }
+}
+
+def Queens() = {
+  var freeMaxs = emptyArray[Boolean]();
+  var freeRows = emptyArray[Boolean]();
+  var freeMins = emptyArray[Boolean]();
+  var queenRows = emptyArray[Int]();
+
+  def getRowColumn(r: Int, c: Int) = {
+    return freeRows.unsafeGet(r) && freeMaxs.unsafeGet(c + r) && freeMins.unsafeGet(c - r + 7);
+  }
+
+  def setRowColumn(r: Int, c: Int, v: Boolean) = {
+    put(freeRows,r, v);
+    put(freeMaxs,c + r, v);
+    put(freeMins,c - r + 7, v);
+  }
+
+  def placeQueen(c: Int): Boolean = {
+    def iterate(r: Int): Return[Boolean] = {
+      if (getRowColumn(r, c)) {
+        put(queenRows,r, c);
+        setRowColumn(r, c, false);
+
+        if (c == 7 || placeQueen(c + 1)) {
```

Comment:
Here we should use two separate `if`s since `||` in Effekt is not lazy.

Reply:
done.
only had to use || because i didnt have early return in the beginning